From 2f97fcc8ea8ab9f8a417daa679d603158f5ea9ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pekka=20J=C3=A4=C3=A4skel=C3=A4inen?=
 <pekka.jaaskelainen@intel.com>
Date: Thu, 15 Dec 2022 19:31:18 +0200
Subject: [PATCH 1/2] Pretend the SPIR ver needed by shuffles is 1.2

In reality the SPIR-V shuffles and ballot operations
require v1.3, but they seem to work with Intel's v1.2
implementation (likely due to the extensions).

Until v1.3 support is added to the driver, let's just pretend
these are v1.2 features to allow CHIPSPV warp-level functions
to compile.

Applies-to: SPIRV-LLVM-Translator llvm_release_230
Upstream-status: deliberate spec deviation, not upstreamable (the
GroupNonUniform* capabilities genuinely require SPIR-V 1.3 per spec).
---
diff --git a/lib/SPIRV/libSPIRV/SPIRVEntry.h b/lib/SPIRV/libSPIRV/SPIRVEntry.h
index e3f9448..ba0e673 100644
--- a/lib/SPIRV/libSPIRV/SPIRVEntry.h
+++ b/lib/SPIRV/libSPIRV/SPIRVEntry.h
@@ -921,7 +921,7 @@ public:
     case CapabilityGroupNonUniformShuffle:
     case CapabilityGroupNonUniformShuffleRelative:
     case CapabilityGroupNonUniformClustered:
-      return VersionNumber::SPIRV_1_3;
+      return VersionNumber::SPIRV_1_2;
 
     case CapabilityFloatControls2:
       return VersionNumber::SPIRV_1_2;
-- 
2.43.0

